compiler/qsc_circuit/src/operations.rs

```diff
@@ -47,6 +47,11 @@ pub enum Error {
 #[must_use]
 pub fn qubit_param_info(item: &Item) -> Option<(Vec<u32>, u32)> {
     if let ItemKind::Callable(decl) = &item.kind {
+        if decl.input.ty == Ty::UNIT {
+            // Support no parameters by allocating 0 qubits.
+            return Some((vec![], 0));
```

Comment:
Return type is an Option. What's the difference between returning None and Some((vec![], 0))?

Reply:
If the callable takes parameters other than qubits, it will return `None` to indicate that we can't generate a circuit.

Reply:
Can you add a comment. This isn't very intuitive.